### PR TITLE
Fix lazy noise (#1886)

### LINKED ISSUE #1886: add_*_noise does not operate lazily for lazy signals
To reproduce the issue:

```python
>>> import hyperspy.api as hs
>>> s = hs.signals.Signal1D([1,2,3]).as_lazy()
>>> s.add_poissonian_noise()
>>> s.data
array([1, 2, 2])
```

### PR DESCRIPTION
**Pull request by francisco-dlp**
### Description of the change

Fixes #1886 and #1894 

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/tests/signal/test_tools.py [hunk 1]
```diff
@@ -184,6 +185,42 @@ def test_numpy_func(self):
...
+            kwargs["chunks"] = s.data.chunks
+        else:
+            data = s.data.copy()
+            from numpy.random import seed, normal
+        seed(1)
+        s.add_gaussian_noise(std=1.0)
+        seed(1)
+        if s._lazy:
+            s.compute()
+        np.testing.assert_array_almost_equal(
```
Comment (ericpre) [inline hyperspy/tests/signal/test_tools.py:209]: If you do `s.compute` for lazy signal, then `s` become a non lazy signal and in the tests there is no need to use the `normal` function from dask?

File: hyperspy/tests/signal/test_tools.py [hunk 2]
```diff
@@ -184,6 +185,42 @@ def test_numpy_func(self):
...
+            kwargs["chunks"] = s.data.chunks
+        else:
+            data = s.data.copy()
+            from numpy.random import seed, poisson
+        seed(1)
+        s.add_poissonian_noise()
+        seed(1)
+        if s._lazy:
+            s.compute()
+        np.testing.assert_array_almost_equal(
```
Comment (ericpre) [inline hyperspy/tests/signal/test_tools.py]: Same as above?

File: hyperspy/tests/signal/test_tools.py (hunk 1 above)
Comment (francisco-dlp) [reply hyperspy/tests/signal/test_tools.py:209]: I also thought so, but it turns out that the random numbers generated by dask and numpy with the same seed differ.

Comment (ericpre) [reply hyperspy/tests/signal/test_tools.py:209]: The tests are failing with the same systematic error. Could it be because of the seed issue?

**Comment by francisco-dlp** [comment]
Thanks @ericpre. Now it should be fixed. I have added a ``keep_dtype`` kwarg to ``add_poissonian_noise`` for convenience. I haven't added it to the gaussian noise one because it is not possible to do it in a safe way, so it's better to let numpy raise an exception rather than having a way to introduce difficult to detect bugs.

**Pull request merged**